src/imars3d/backend/workflow/engine.py

```diff
@@ -37,17 +41,17 @@ def _validate_outputs(task_outputs: Iterable[Any], function_outputs: Optional[It
         """
 
         def validate_type(outputs):
-            if not hasattr(outputs, "__iter__"):
-                raise WorkflowEngineError("Task and Function outputs must be iterable")
-            if isinstance(outputs, str):
-                raise WorkflowEngineError("Task and Function outputs cannot be string")
+            if not isinstance(outputs, (list, tuple)):
+                raise WorkflowEngineError("Task outputs must be a list or a tuple")
 
         validate_type(task_outputs)
-        if function_outputs:
-            validate_type(function_outputs)
+        if function_outputs is not None:
+            if type(function_outputs) is not tuple:
+                function_outputs = (function_outputs,)
```

Comment:
This is not supposed to happen according to the JSON schema. We agreed that `function_outputs` is either `None` or a `tuple`. If `function_outputs` is needed, then we should `raise WorkflowEngineError` instead of "fixing it"